app/assets/src/components/utils/extractChildren.js

```diff
@@ -0,0 +1,35 @@
+import React from "react";
+import { clone, difference, filter } from "lodash";
+
+// Separates the children based on type.
+// children - this.props.children.
+// componentTypes - custom strings that must be manually added to sub-components.
+//   ex: static type = "ViewHeader.Content"
+// Returns an object that maps types to the matching component, and returns unmatched compoents in 'other'.
+const extractChildren = (children, componentTypes) => {
```

Comment:
I wonder if we can turn `extractChildren` into a cutomizable proptypes function checker?

Reply:
Holding off for now. Custom prop validators don't appear to be able to take custom arguments...Seems non-trivial to implement.